src/lexer/regexFactory.ts

```diff
@@ -165,7 +170,9 @@ export const identifierPattern = ({ first, rest, dashes }: IdentChars = {}): str
   const firstChars = escapeRegExp(first ?? '');
   const restChars = escapeRegExp(rest ?? '');
 
-  const pattern = `[${letter}${firstChars}][${letter}${number}${restChars}]*`;
+  const pattern = allowFirstCharNumber
+    ? `[${firstChars}]?[${letter}${number}${restChars}]*[${letter}${restChars}]+[${letter}${number}${restChars}]*`
```

Comment:
I don't understand this regex. What is it trying to achieve?

Reply:
matches any amount of letter|number|other char with at least 1 non-number char

Reply:
So, according to this regex, when first character can be `_` we allow `_abc` as identifier, but `_123` is invalid.

I think it would be better to not tackle this problem in here at all. Let's just rely on the order of matching:

- we first try to match a number
- when it succeeds, we won't be attempting to match this same text as identifier.
- when it fails, we know we're not dealing with the invalid case of digits-only identifier.

That's IMHO how most lexers deal with similar issues.